src/Indicator.vala

```diff
@@ -81,23 +80,27 @@ public class DateTime.Indicator : Wingpanel.Indicator {
         return main_grid;
     }
 
-    private void create_event_grid () {
-        event_grid = new Gtk.Grid ();
-        event_grid.orientation = Gtk.Orientation.VERTICAL;
-        main_grid.attach (event_grid, 0, 1);
+    private void update_events_model (E.Source source, Gee.Collection<E.CalComponent> events) {
+        idle_update_events ();
     }
 
-    private void update_events_model (E.Source source, Gee.Collection<E.CalComponent> events) {
-        Idle.add (update_events);
+    private void idle_update_events () {
+        if (update_events_idle_source > 0) {
+            Source.remove (update_events_idle_source);
+            update_events_idle_source = 0;
+        }
+        update_events_idle_source = Idle.add (update_events);
     }
 
     private bool update_events () {
-        event_grid.destroy();
+        if (event_grid != null) event_grid.destroy();
```

Comment:
Code style: No single line if clauses without braces. Always use braces and multiple lines.

Reply:
We should clarify this at: https://elementary.io/docs/code/reference#reference